### PR TITLE
Fix characters equals trim

### PR DESCRIPTION
**Pull request by VLysak**

### CODE REVIEW

**Review by vkushni — CHANGES_REQUESTED** [review]
please rebase branch to develop

File: src/main/java/org/util/string/StringUtil.java [hunk 1]
```diff
@@ -836,23 +838,23 @@ else if (obj == null || builder.getClass() != obj.getClass() || obj.toString().l
     // this method group uses default
     // type
     public boolean equals(String value) {
-        return builder.equals(value);
+        return toString().equals(value);
     }
 
     public boolean equals(char[] value) {
-        return builder.equals(value);
+        return toString().equals(new String(value));
```
Comment (VLysak) [inline src/main/java/org/util/string/StringUtil.java]: 1

File: src/main/java/org/util/string/StringUtil.java [hunk 2]
```diff
@@ -836,23 +838,23 @@ else if (obj == null || builder.getClass() != obj.getClass() || obj.toString().l
...
     }
 
     public boolean equals(StringBuilder value) {
-        throw new UnsupportedOperationException();
+        return toString().equals(value.toString());
```
Comment (VLysak) [inline src/main/java/org/util/string/StringUtil.java]: 1

File: src/main/java/org/util/string/StringUtil.java [hunk 3]
```diff
@@ -836,23 +838,23 @@ else if (obj == null || builder.getClass() != obj.getClass() || obj.toString().l
...
     }
 
     public boolean equals(StringBuffer value) {
-        throw new UnsupportedOperationException();
+        return toString().equals(value.toString());
```
Comment (VLysak) [inline src/main/java/org/util/string/StringUtil.java]: 1

File: src/main/java/org/util/string/StringUtil.java [hunk 4]
```diff
@@ -836,23 +838,23 @@ else if (obj == null || builder.getClass() != obj.getClass() || obj.toString().l
...
     }
 
     public boolean equals(StringUtil value) {
-        throw new UnsupportedOperationException();
+        return toString().equals(value.toString());
```
Comment (VLysak) [inline src/main/java/org/util/string/StringUtil.java]: 1

File: src/main/java/org/util/string/StringUtil.java [hunk 5]
```diff
@@ -875,12 +877,37 @@ public boolean equals(StringUtil value, Equality equality) {
         throw new UnsupportedOperationException();
     }
 
-    public StringUtil trim() {
-        return trim(Trim.BOTH);
-    }
+//    public StringUtil trim() {
```
Comment (VLysak) [inline src/main/java/org/util/string/StringUtil.java]: 111

File: src/main/java/org/util/string/StringUtil.java [hunk 6]
```diff
@@ -875,12 +877,37 @@ public boolean equals(StringUtil value, Equality equality) {
...
     public StringUtil trim(Trim trim) {
-        throw new UnsupportedOperationException();
+
+        String str = "";
+        char[] charArray = new char[size()];
+
+        builder.getChars(0, size(), charArray, 0);
+        if (trim == Trim.BOTH) {
+            str = toString().trim();
+        } else if (trim == Trim.LEADING) {
```
Comment (VLysak) [inline src/main/java/org/util/string/StringUtil.java:892]: 1

**Review by vkushni — APPROVED** [review]

**Pull request merged**